art/attacks/evasion/pixel_threshold.py

```diff
@@ -0,0 +1,1457 @@
+                opts.set('verb_disp', 40000)
+                opts.set('verb_log', 40000)
+                opts.set('verb_time', False)
+
+            opts.set('bounds', bounds)
+
+            if self.type_attack == 0:
+                std = 63
+            else:
+                std = limit
+
+            strategy = CMAEvolutionStrategy(initial, std / 4, opts)
+
+            try:
+                strategy.optimize(
+                    predict_fn,
+                    maxfun=max(
+                        1,
+                        400 //
+                        len(bounds)) *
+                    len(bounds) *
+                    100,
+                    callback=callback_fn)
+            except Exception as exception:
+                if self.verbose:
+                    print(exception)
+                pass
+
+            adv_x = strategy.result[0]
+
+        else:
+
+            strategy = differential_evolution(
+                predict_fn,
+                bounds,
+                disp=self.verbose,
+                maxiter=100,
+                popsize=max(
+                    1,
+                    400 // len(bounds)),
+                recombination=1,
+                atol=-1,
+                callback=callback_fn,
+                polish=False)
+            adv_x = strategy.x
+
+        if self._attack_success(adv_x, image, target_class):
+            return True, self._perturb_image(adv_x, image)[0]
+        else:
+            return False, image
+
+
+class PixelAttack(PixelThreshold):
+    """
+    This attack was originally implemented by Vargas et al. (2019).
+    It is generalisation of One Pixel Attack originally implemented by
+    Su et al. (2019)
+
+    | One Pixel Attack Paper link:
+        https://ieeexplore.ieee.org/abstract/document/8601309/citations#citations
+        (arXiv link: https://arxiv.org/pdf/1710.08864.pdf)
+    | Pixel Attack Paper link:
+        https://arxiv.org/abs/1906.06026
+    """
+
+    def __init__(
+            self,
+            classifier,
+            th=None,
+            es=0,
+            targeted=False,
+            verbose=False):
+        """
+        Create a :class:`.PixelAttack` instance.
+        """
+        super(
+            PixelAttack,
+            self).__init__(
+                classifier,
+                th,
+                es,
+                targeted,
+                verbose)
+        self.type_attack = 0
+
+    def _perturb_image(self, x, img):
+        """
+        Perturbs the given image `img` with the given perturbation `x`.
+        """
+        if x.ndim < 2:
+            x = np.array([x])
+        imgs = np.tile(img, [len(x)] + [1] * (x.ndim + 1))
+        x = x.astype(int)
+        for adv, image in zip(x, imgs):
+            for pixel in np.split(adv, len(adv) // (2 + image.shape[-1])):
+                x_pos, y_pos, *rgb = pixel
+                image[x_pos % image.shape[-3], y_pos % image.shape[-2]] = rgb
+        return imgs
```

Comment:
As requested, this is the first instance where input image shape is used. 
This function adds the perturbation to the image. 
Please note that the Pixel Attack perturbation is a vector of 5 variables for 3 channel image and 3 variable for 1 channel image, 
`For 3 Channel Index: <Position of X>, <Position of Y>, <Value of Red Channel>, <Value of Blue Channel>, <Value of Green Channel>`
`For 1Channel Index: <Position of X>, <Position of Y>, <Value of Monochrome Channel>'.

I tried to use `classifer.input_shape` but this parameter is not mandatory for all classifiers, therefore, I did not use it.

Reply:
If you would use `classifer.input_shape`, could you avoid `swapaxes`?

Reply:
Indeed, if I initially have `classifer.input_shape`, I could definitely avoid `swapaxes`.